src/parser/doesSomeParentMatch.ts

```diff
@@ -1,6 +1,6 @@
 import { selectorMatches } from './selectorMatches';
 
-export function doesSomeParentMatch(element: HTMLElement, selectors: string[]) {
+export function doesSomeParentMatch(element: HTMLElement | SVGElement, selectors: string[]) {
```

Comment:
I extended this type on some places because parser tests are passing SVGElement to those functions
and if function accepts it and can deal with it then types should reflect that
this way we have no type issues in the tests and also tests reflect that back to the user

Reply:
this was written for svgElements, i m not sure why HTMLElement ended up there. But anyway this is correct. If some types ends up being strange they are probably wrong, like shadow that had options not optional but mandatory and you had to pass an empty object to initialize a standard shadow.